python/src/deltachat/account.py

```diff
@@ -266,5 +410,13 @@ def convert_to_bytes_utf8(obj):
     return obj
 
 
+def iter_array_and_unref(dc_array_t, constructor):
+    try:
+        for i in range(0, lib.dc_array_get_cnt(dc_array_t)):
+            yield constructor(lib.dc_array_get_id(dc_array_t, i))
```

Comment:
Is the underlying deltachat data here actually a fully populated array with all data or an array of IDs/pointers or an iterator?  I ask because if it's not the first it might be worth creating an DCArrayIterator object or so which is actually a real iterator (implementing collections.abc.Iterator), or maybe a DCArrayContainer if it's indexable or can be iterated multiple times (implementing collections.abc.Collection or whatever is appropriate)

Reply:
it's an fully populated array, so `dc_array_get_id()` is cheap, only a function call and a return a[i], no i/o is involved here.
typical arrays have some 1000 entries (eg. a longer chat history) most are much smaller, typically not one million or so :)